src/PowerShellEditorServices/Services/PowerShellContext/PowerShellContextService.cs

```diff
@@ -2102,25 +2102,24 @@ private void SetExecutionPolicy()
             // - Process
             // - CurrentUser
             // - LocalMachine
-            // This is the order of precedence we want to follow, skipping the Process scope
+            // We want to ignore policy settings, since we'll already have those anyway.
+            // Then we need to look at the CurrentUser setting, and then the LocalMachine setting.
             //
             // Get-ExecutionPolicy -List emits PSObjects with Scope and ExecutionPolicy note properties
             // set to expected values, so we must sift through those.
+
             ExecutionPolicy policyToSet = ExecutionPolicy.Bypass;
-            for (int i = policies.Count - 1; i >= 0; i--)
+            var currentUserPolicy = (ExecutionPolicy)policies[policies.Count - 2].Members["ExecutionPolicy"].Value;
```

Comment:
I like this, can we rely on the order to remain consistent though?

Reply:
I [*think* so](https://github.com/PowerShell/PowerShell/blob/69bf7043c0c7471679265778e12053c1c40b0000/src/System.Management.Automation/security/SecuritySupport.cs#L121-L133)

Reply:
The other alternative is to look for user scope and then to look for machine scope

Reply:
But I don't think there's any plan to update anything around execution policy